### PR TITLE
chore(deps): upgrade node.js version to v16

### PR DESCRIPTION
**Pull request by yakkomajuri**
## Changes

👉 *Stay up-to-date with our [coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

It's about time we upgraded to the latest node LTS version.

This should be better overall but might also help us get rid of some V8 bugs we've identified.

### CODE REVIEW

**Comment by yakkomajuri** [comment]
Added you @guidoiaquinti for any considerations I might have missed when updating docker files

File: plugin-server/src/main/pluginsServer.ts [hunk 1]
```diff
@@ -119,6 +119,12 @@ export async function startPluginsServer(
         process.exit(0)
     })
 
+    process.on('unhandledRejection', (error: Error) => {
```
Comment (yakkomajuri) [inline plugin-server/src/main/pluginsServer.ts:122]: @mariusandra here I'm doing the same thing the workers do but also on the main thread. We void things here too and wouldn't want to crash if those throw

**Comment by yakkomajuri** [comment]
@guidoiaquinti @mariusandra can I merge this in? Want to get it early in the day before the spike and so that I can monitor it all day

**Review by mariusandra — APPROVED** [review]
I'd say go for it... with an open question: do we want this change in the next release, or do we want to wait? If the first option, we need to communicate the change well in the release notes as well.

**Comment by yakkomajuri** [comment]
@mariusandra fair point - I think waiting makes sense, because that ensures our release notes are not too overloaded (they will already have some async migrations warnings).

_However_, I'd like to merge and maybe revert by the end of the day as I'd like to know if this solves the cryptic V8 problem or if that needs more attention

**Review by guidoiaquinti — APPROVED** [review]
LGTM

**Pull request merged**